WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -301,27 +302,28 @@ private extension ProductSelectorView {
                 ServiceLocator.analytics.track(event: .ProductListFilter.productListViewFilterOptionsTapped(source: source.filterAnalyticsSource))
             }
             .buttonStyle(LinkButtonStyle())
-            .fixedSize()
         }
-        .padding(.horizontal, insets: safeAreaInsets)
         .background(Color(.listForeground(modal: false)))
     }
 
     @ViewBuilder private var productSelectorHeaderSearchRow: some View {
-        HStack {
-            SearchHeader(text: $viewModel.searchTerm, placeholder: Localization.searchPlaceholder, onEditingChanged: { isEditing in
-                searchHeaderisBeingEdited = isEditing
-            })
-            .padding(.horizontal, insets: safeAreaInsets)
-            .accessibilityIdentifier("product-selector-search-bar")
-            Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
-                ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+        GeometryReader { geometry in
+            HStack {
+                SearchHeader(text: $viewModel.searchTerm, placeholder: Localization.searchPlaceholder, onEditingChanged: { isEditing in
+                    searchHeaderisBeingEdited = isEditing
+                })
+                .accessibilityIdentifier("product-selector-search-bar")
+                Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
+                    ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+                }
+                .if(horizontalSizeClass == .regular && geometry.size.width < Constants.headerSearchRowWidth) { $0.pickerStyle(.menu) }
+                .if(horizontalSizeClass == .compact || geometry.size.width > Constants.headerSearchRowWidth) { $0.pickerStyle(.segmented) }
```

Comment:
Depending how much we perfect it, some of this could be usefully moved in to a new generic view similar to AdaptiveStack. But I think it might be easier to simplify the design so we're not cramming so much in, and then rely on default stack behaviour.